cassovary-core/src/main/scala/com/twitter/cassovary/util/collections/FastQueue.scala

```diff
@@ -0,0 +1,119 @@
+package com.twitter.cassovary.util.collections
+
+import it.unimi.dsi.fastutil.PriorityQueue
+import it.unimi.dsi.fastutil.ints.{IntComparator, IntHeapPriorityQueue, IntArrayFIFOQueue, IntPriorityQueue}
+import it.unimi.dsi.fastutil.objects.{ObjectArrayFIFOQueue, ObjectArrayPriorityQueue}
+
+abstract class FastQueue[@specialized(Int, Long) T] {
```

Comment:
Do you think we can just do FastMap as first step, then move to FastQueue in another PR. This PR is going to be big anyway.